### PR TITLE
Ask "Does this excrement bag dispenser have a bin?"

### LINKED ISSUE #4370: Quest suggestion: Does this poop bag dispenser have a trash can?
### General
Affected tag(s) to be modified/added: [`bin`](https://wiki.openstreetmap.org/wiki/Key:bin)
Question asked: **Does this poop bag dispenser have a trash can?**

### Checklist
Checklist for quest suggestions (see [guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md)):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by any pedestrian from the outside but a survey is necessary
- [x] 💤 Not an overwhelming percentage of quests have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of map data (Worth the effort)

### Links

* https://taginfo.openstreetmap.org/tags/vending=excrement_bags#combinations
* https://wiki.openstreetmap.org/wiki/Tag%3Avending%3Dexcrement_bags

### Ideas for implementation

**Element selection:**

```
nodes with
  vending = excrement_bags
  and (!bin or (bin = no and bin older today -12 years))
```

**Metadata needed:** None

**Proposed UI:** Simple Yes/No quest

### PR DESCRIPTION
**Pull request by qugebert**
Closes #4370

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/quests/excrement_bag_dispenser/AddExcrementBagDispenserHasBin.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          and access !~ private|no
+    """.toElementFilterExpression() }
+
+    private val nearbyBinsFilter by lazy { """
+        nodes with
+          (
+          amenity = waste_basket
+          or bin = yes
+          )
+          and access !~ private|no
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/quests/excrement_bag_dispenser/AddExcrementBagDispenserHasBin.kt:31]:
```suggestion
        nodes with
          (amenity = waste_basket or bin = yes)
          and access !~ private|no
```

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1043,6 +1043,8 @@ Before uploading your changes, the app checks with a &lt;a href=\"https://www.we
 
     <string name="quest_gritBinSeasonal_title">Is this also here outside of winter?</string>
 
+    <string name="quest_excrement_bag_dispenser_bin_title">Does this excrement bag dispenser have a bin?</string>
```
Comment (FloEdelmann) [inline app/src/main/res/values/strings.xml:1046]: Maybe being a bit more colloquial is okay/preferable?

```suggestion
    <string name="quest_excrement_bag_dispenser_bin_title">Does this poop bag dispenser have a bin?</string>
```

Comment (qugebert) [reply app/src/main/res/values/strings.xml:1046]: Sure, this is fine for me too

**Comment by westnordost** [comment]
Hm sorry, the linked issue ticket slipped under my radar. Generally, only features and new quests with `new quest` or `enhancement` have the "stamp of approval" in the sense that all doubts have been cleared and a PR would be accepted. I will look through the linked issue now and comment there, first.

**Comment by westnordost** [comment]
This PR can stay open until the discussion is over. However, I would not work on it further until then.

**Pull request closed without merging**